topi/python/topi/nn/util.py

```diff
@@ -0,0 +1,101 @@
+# pylint: disable=invalid-name, unused-variable
+"""NN operator common utilities"""
+from __future__ import absolute_import
+
+def infer_pad(data, data_pad):
+    """Infer the padding from stages in reverse.
+
+    Parameters
+    ----------
+    data : Tensor
+        data stage.
+
+    data_pad : Tensor
+        pad stage.
+
+    Returns
+    -------
+    hpad : int
+        padding size on height
+    wpad : int
+        padding size on width
+    """
+    if data_pad is None:
+        return 0, 0
+    _, _, IH, IW = [x.value for x in data.shape]
+    _, _, TH, TW = [x.value for x in data_pad.shape]
+    hpad = (TH - IH) // 2
+    wpad = (TW - IW) // 2
+    return hpad, wpad
+
+def infer_stride(data, kernel, out):
+    """Infer the stride from stages in reverse.
+
+    Parameters
+    ----------
+    data : Tensor
+        data stage.
+
+    kernel : Tensor
+        kernel stage.
+
+    out : Tensor
+        output stage.
+
+    Returns
+    -------
+    hstride : int
+        stride size on height
+    wstride : int
+        stride size on width
+    """
+    _, _, IH, IW = [x.value for x in data.shape]
+    _, _, KH, KW = [x.value for x in kernel.shape]
+    _, _, OH, OW = [x.value for x in out.shape]
```

Comment:
Make it work for non constant symbolic value, add a testcase for this

Reply:
if the workload is symbolic, we have no preserved schedule for it